.github/workflows/verifyHybridApp.yml

```diff
@@ -0,0 +1,141 @@
+name: Verify HybridApp build
+
+on:
+  workflow_call:
+  pull_request:
+    types: [opened, synchronize]
+    branches-ignore: [staging, production]
+    paths:
+      - '**.kt'
+      - '**.java'
+      - '**.swift'
+      - '**.mm'
+      - '**.h'
+      - '**.cpp'
+      - 'package.json'
+      - 'patches/**'
+      - 'android/build.gradle'
+      - 'android/AndroidManifest.xml'
+      - 'ios/Podfile.lock'
+      - 'ios/project.pbxproj'
+
+concurrency:
+  group: ${{ github.ref == 'refs/heads/main' && format('{0}-{1}', github.ref, github.sha) || github.ref }}-verify-main
+  cancel-in-progress: true
+
+jobs:
+  verify_android:
+    name: Verify Android HybridApp builds on main
+    runs-on: ubuntu-latest-xl
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+          ref: ${{ github.event.pull_request.head.sha }}
+          token: ${{ secrets.OS_BOTIFY_TOKEN }}
+          # fetch-depth: 0 is required in order to fetch the correct submodule branch
+          fetch-depth: 0
+
+      - name: Update submodule to match main
+        run: |
+          git submodule update --init --remote
+          git fetch
+          git checkout main
```

Comment:
Here, you're checking out the main branch of E/App, right? Is that what we want?

Reply:
Will include in follow up